ballerina/graphql_client.bal

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+import ballerina/http;
+ 
+# The [Ballerina](https://ballerina.io/) generic client for GraphQL(https://graphql.org/) APIs.
+public isolated client class Client {
+    final http:Client httpClient;
+
+    # Gets invoked to initialize the `connector`.
+    #
+    # + serviceUrl - URL of the target service
+    # + clientConfig - The configurations to be used when initializing the `connector`
+    # + return - An error at the failure of client initialization
+    public isolated function init(string serviceUrl, http:ClientConfiguration clientConfig = {})  returns ClientError? {
+        http:Client|http:ClientError httpClient = new (serviceUrl, clientConfig);
+        if httpClient is http:ClientError {
+             return error RequestError("GraphQL Client Error", httpClient);
+        }
+        self.httpClient = httpClient;
+    }
+
+    # Executes a GraphQL query and data binds the GraphQL response to a record with data and extensions 
+    # which is a subtype of GenericResponse.
+    #
+    # + document - The GraphQL query or mutation. For example `query countryByCode($code:ID!) {country(code:$code) {name}}`.
```

Comment:
How does this handle documents with multiple operations? A GraphQL document can have more than one operation. GraphQL request should contain the operation name in such scenarios. We should handle this.

```suggestion
    # + document - The GraphQL document. For example `query countryByCode($code:ID!) {country(code:$code) {name}}`.
```

Reply:
The `operationName` is added as a new optional parameter in the client functions & test cases were added in [f26678c](https://github.com/ballerina-platform/module-ballerina-graphql/pull/663/commits/f26678c4ceef61a6ce293851142e167fd6e26c07). Thank you.